modules/account-lib/src/coin/baseCoin/iface.ts

```diff
@@ -130,3 +130,29 @@ export type FeeOptions = {
   formula?: 'fixed' | 'feeRate' | 'perKB' | 'custom';
   type?: 'base' | 'max' | 'tip';
 };
+
+/**
+ * The validity window signals when a transaction can be included in the network.
+ *  ValidityWindow information is stored as part of the transaction data in the indexer and wallet platform, unless it is absolutely necessary.
+ */
+export interface ValidityWindow {
```

Comment:
Not sure what's the best place to add this, but we should also have methods that validate constraints: 

firstValid must be set and will default to timestamp/blockheight if not specified
lastValid is optional and is validated or set as follows
- if lastValid, firstValid + minDuration <= lastValid <= firstValid + maxDuration && firstValid < lastValid
- if not lastValid, lastValid = firstValid + minDuration // transaction valid for the coin's minimum validity window by default if lastValid not specified

minDuration < maxDuration must be validated

This will make sure that firstValid and lastValid are always defined for a transaction.

Reply:
Sounds good: these methods will be covered here:
https://bitgoinc.atlassian.net/browse/STLX-10800